### DISCUSSION TITLE
How to detect ```CC(C)OO``` but not ```CC(C)O[O]```

### DISCUSSION THREAD

**Discussion by tsn2027** [rdkit/rdkit#6551] [Q&A] [answered]
Hello,

I have 2 structures ```CC(C)O[O]``` (which has a radical on Oxygen) and ```CC(C)OO``` (which does not have any radicals). How can I either use the ```HasSubstructMatch()``` function or write a customized function to detect ```CC(C)OO``` but not ```CC(C)O[O]```? I could find a structure which has a radical by using ```HasSubstructMatch``` (which is mentioned [here](https://www.rdkit.org/docs/RDKit_Book.html#cxsmiles) but not another way around. Thus, I am looking for helps. Thanks in advance.

**Comment by ptosco** [comment, marked as answer]
You can use two SMARTS expressions:
`CC(C)O[OH1]` for the peroxide and `CC(C)O[OH0]` for the radical.

**Reply by tsn2027** [reply]
Thank you! This is what I am looking for!
